client/actions.js

```diff
@@ -0,0 +1,308 @@
+            })
+        })
+      }
+    if (searchArray === 2) {
+      fetch(`/searchnames/${searchArray[0]}/${searchArray[1]}`, {
+        headers: {'Content-Type': 'application/json'}
+      })
+        .then(response => response.json())
+        .then(response => {
+          dispatch(renderResults(response))
+        })
+    }
+
+  }
+}
+
+const employeeSubmitted = () => {
+  return { type: 'EMPLOYEE_SUBMITTED' }
+}
+
+const employeeSaved = (response) => {
+  return { type: 'EMPLOYEE_SAVED', response }
+}
+
+const employeeFailure = () => {
+  return { type: 'EMPLOYEE_FAILURE' }
+}
+
+const formCleared = () => {
+  return { type: 'FORM_CLEARED' }
+}
+
+const saveEmployee = employee => {
+  return dispatch => {
+    dispatch(employeeSubmitted());
+
+    fetch('/newemployee/', {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(employee)
+    })
+      .then(response => response.json())
+      .then(response => {
+        if (response.error) {
+          alert(response.error);
+          dispatch(employeeFailure());
+        }
+        else if (response.id) {
+          alert(`Employee ${response.id} ${response.first_name} ${response.last_name} succcessfully created!`)
+          dispatch(employeeSaved(response));
+        }
+
+      })
+      .then( () => {
+        if (document.getElementById('employee')) {
+          document.getElementById('employee').reset();
+          dispatch(formCleared());
+        }
+      })
+  }
+}
+
+const idSearch = () => {
+  return { type: 'ID_SEARCH' }
+}
+
+const idFound = (response) => {
+  return { type: 'ID_FOUND', response }
+}
+
+const idNotFound = () => {
+  return { type: 'ID_NOT_FOUND' }
+}
+
+const idCleared = () => {
+  return { type: 'ID_CLEARED' }
+}
+
+const renderProfile = employeeId => {
+  return dispatch => {
+    dispatch(idSearch());
+
+    fetch(`/viewemployee/${employeeId}`, {
+      headers: {'Content-Type': 'application/json'}
+    })
+      .then(response => response.json())
+      .then(response => {
+        if (response.error) {
+          alert(response.error);
+          dispatch(idNotFound());
+        }
+        else if (response.id) {
+          dispatch(idFound(response))
+        }
+      })
+      .then( () => {
+        if (document.getElementById('find-id')) {
+          document.getElementById('find').reset();
+          dispatch(idCleared());
+        }
+      })
+
+  }
+}
+
+const editForm = (response) => {
+  return { type: 'EDIT_FORM', response }
+}
+
+const readyToEdit = () => {
+  return { type: 'READY_TO_EDIT'}
+}
+
+const updateProfile = employeeId => {
+  return dispatch => {
+    dispatch(idSearch());
+
+    fetch(`/viewemployee/${employeeId}`, {
+      headers: {'Content-Type': 'application/json'}
+    })
+      .then(response => response.json())
+      .then(response => {
+        dispatch(editForm(response));
+        document.getElementById('employee-id').value = response.id;
```

Comment:
Woah, you should totally be connecting to the `store` in your form component to get these values.